mlir/include/mlir/Tools/mlir-opt/MlirOptMain.h

```diff
@@ -62,6 +62,17 @@ class MlirOptMainConfig {
     return allowUnregisteredDialectsFlag;
   }
 
+  /// Load all available dialects in the context.
+  /// This option is for convenience during testing only and discouraged in
```

Comment:
This entire tool is only for testing. Can you elaborate on what tests are breaking without doing this? This was removed because it is almost always a bug if passes are not declaring their dependent dialects in a way that would require this.